benchmark/src/txn_generator.rs

```diff
@@ -124,53 +167,99 @@ pub fn gen_mint_txn_requests(
-/// transfer. For example, given account (A1, A2, A3, ..., AN), this method returns a vector
-/// of TXNs like (A1->A1, A1->A2, ..., A1->AN, A2->A1, A2->A2, ... A2->AN, ..., AN->A(N-1)).
-pub fn gen_pairwise_transfer_txn_requests(
-    txn_signer: &WalletLibrary,
-    accounts: &mut [AccountData],
-) -> Vec<SubmitTransactionRequest> {
-    let receiver_addrs: Vec<AccountAddress> =
-        accounts.iter().map(|account| account.address).collect();
-    let mut txn_reqs = vec![];
-    for sender in accounts.iter_mut() {
-        for receiver_addr in receiver_addrs.iter() {
-            match gen_transfer_txn_request(sender, receiver_addr, txn_signer, 1) {
-                Ok(txn_req) => txn_reqs.push(txn_req),
-                Err(e) => {
+impl RingTransferTxnGenerator {
+    pub fn new() -> Self {
+        let wallet = WalletLibrary::new();
+        RingTransferTxnGenerator { wallet }
+    }
+}
+
+impl LibraRequestGenerator for RingTransferTxnGenerator {
+    fn gen_accounts(&mut self, num_accounts: u64) -> Vec<AccountData> {
+        gen_accounts_from_wallet(&mut self.wallet, num_accounts)
+    }
+    fn gen_setup_txn_requests(
+        &self,
+        faucet_account: &mut AccountData,
+        accounts: &mut [AccountData],
+    ) -> Vec<LibraRequest> {
+        gen_mint_txn_requests(faucet_account, accounts)
+    }
+    fn gen_signed_txn_request_load(&self, accounts: &mut [AccountData]) -> Vec<LibraRequest> {
+        let mut receiver_addrs: Vec<AccountAddress> =
+            accounts.iter().map(|account| account.address).collect();
+        receiver_addrs.rotate_left(1);
+        accounts
+            .iter_mut()
+            .zip(receiver_addrs.iter())
+            .flat_map(|(sender, receiver_addr)| {
+                gen_transfer_txn_request(sender, receiver_addr, &self.wallet, 1).or_else(|e| {
                     error!(
                         "failed to generate {:?} to {:?} transfer TXN: {:?}",
                         sender.address, receiver_addr, e
                     );
+                    Err(e)
+                })
+            })
+            .collect()
+    }
+}
+
+/// Pre-generate TXN requests of pairwise transfers between accounts, including self to self
+/// transfer. For example, given account (A1, A2, A3, ..., AN), this method returns a vector
+/// of TXNs like (A1->A1, A1->A2, ..., A1->AN, A2->A1, A2->A2, ... A2->AN, ..., AN->A(N-1)).
+pub struct PairwiseTransferTxnGenerator {
+    /// Use WalletLibrary to generate accounts and sign transfer TXNs.
+    wallet: WalletLibrary,
+}
+
+impl PairwiseTransferTxnGenerator {
+    pub fn new() -> Self {
+        let wallet = WalletLibrary::new();
+        PairwiseTransferTxnGenerator { wallet }
+    }
+}
+
+impl LibraRequestGenerator for PairwiseTransferTxnGenerator {
+    fn gen_accounts(&mut self, num_accounts: u64) -> Vec<AccountData> {
+        gen_accounts_from_wallet(&mut self.wallet, num_accounts)
+    }
+    fn gen_setup_txn_requests(
```

Comment:
nit: add an empty line between functions to be consistent with the rest of the codebase.

Reply:
Done